dre-service/src/main/java/io/cdap/re/NamespacedId.java

```diff
@@ -0,0 +1,65 @@
+/*
+ *
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.re;
+
+import java.util.Objects;
+
+/**
+ * A unique identifier for an entity within a namespace. The 'id' field is not globally unique, but is unique within the
+ * context. It is named 'id' for backward compatibility reasons.
```

Comment:
is 'id' used in any REST requests or responses? In wrangler it was, which is why 'id' was kept. But if this is an internal name in the rules engine code, then it should be renamed.

Reply:
Yes, it is. For example, it is used when creating rules and rulebooks, retrieving a specific rule or rulebook, and updating a specific rule and rulebook. Are those the types of usages of requests and responses that you meant?

Reply:
I mean HTTP requests and responses. Like if getting a rule results in a json response like:

```
{
  "id": "abc123",
  ...
}
```

Reply:
Yes, it does